### PR TITLE
Fix some static analyser fixies

### PR DESCRIPTION
**Pull request by asapelkin**
##  Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change which fixes an issue)

## Description and Context
Some cppcheck and clang-tidy fixies (and a bit from me)
 
## Testing strategy
By existing tests

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

**Comment by asapelkin** [comment]
rebased onto 2.5.1

File: tests/TestCsvParser.cpp [hunk 1]
```diff
@@ -111,7 +111,7 @@ void TestCsvParser::testEmptySimple()
     out << "";
     QVERIFY(parser->parse(file.data()));
     t = parser->getCsvTable();
-    QVERIFY(t.size() == 0);
+    QVERIFY(t.empty());
```
Comment (phoerious) [inline tests/TestCsvParser.cpp]: These should be isEmpty(), since we are writing Qt.

Comment (droidmonkey) [reply tests/TestCsvParser.cpp]: Yah I believe empty() is similar to clear(). Great catch

Comment (asapelkin) [reply tests/TestCsvParser.cpp]: No, empty is not  similar to clear. 
IsEmpty() is equivalent for empty(), but more native for qt, OK.

**Pull request merged**